components/service/glean/src/main/java/mozilla/components/service/glean/storages/EventsStorageEngine.kt

```diff
@@ -169,15 +338,41 @@ internal object EventsStorageEngine : StorageEngine {
 
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     override fun clearAllStores() {
+        // We need to delete these files on the IO thread to avoid
+        // racing with writing to the file.  However, we want to
+        // block the main thread until this work is done, so the
+        // caller can rely on the files being gone.  This is not a
+        // performance problem since this function is for use in
+        // testing only.
```

Comment:
There's no single IO thread. `Dispatchers.IO` uses at least 64 threads. Can this be solved with other synchronization mechanisms?

Reply:
Ah, thanks.  I think using a `synchronized(lock) { }` should work just fine, then.